### PR TITLE
Add some weekly VPNKit development reports

### PR DESCRIPTION
**Pull request by avsm**
These reports have been generated from GitHub activity, with some
hand editing in the style of the LinuxKit reports. They can be
browsed directly within the repository and cover DataKit and
immediately related repositories such as ocaml-git and 9p. Fixes
can be contributed directly to this repository.

For the brave who want to generate them for their own repositories,
the work-in-progress generator of the skeleton of these reports
is in https://github.com/avsm/hesternus

Should I add any more repos to track as part of this weekly series?
cc @magnuss @yomimono @djs55 @dave-tucker @balrajsingh

### CODE REVIEW

**Comment by yomimono** [comment]
I think [dns-forward](https://github.com/djs55/ocaml-dns-forward) should be included too.

**Comment by yomimono** [comment]
The dating is odd here; can the filenames have dates representing the last day in the covered period rather than the first?

The combination of the 24th April report and the 1st May remind me that there are things from last week left undone, which is a good thing for these reports to do.

File: README.md [hunk 1]
```diff
@@ -36,3 +37,14 @@ respect to the host's VPN configuration.
 VPNKit operates by reconstructing Ethernet traffic from the VM and translating it into the
 relevant socket API calls on OSX or Windows. This allows the host application to generate
 traffic without requiring low-level Ethernet bridging support.
+
+Licensing
+---------
+
+DataKit is licensed under the Apache License, Version 2.0. See
```
Comment (djs55) [inline README.md]: s/DataKit/VPNKit/?

Comment (avsm) [inline README.md]: fixed in 2a3a6bd61142d90743e94246aa71c3fa7c3dc9ef

**Comment by avsm** [comment]
@djs55 I have pushed the latest dev reports; if you could review and merge I'd appreciate it.

File: reports/2017-05-28.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+Scalability of VPNKit was also improved for a large number of connections,
+and is now configurable via the `--listen-backlog <n>` flag.
+
+Various spurious DNS failures were also fixed, leading to the release of
+[mirage/ocaml-dns:0.20.1].
+
+One area where we could use contributions in TCP/IP is with the test suite,
+to help spot regressions in performance and retransmission behaviour.  Please
+see [mirage/mirage-tcpip#140] or [mirage/mirage-tcpip#314] for more information.
```
Comment (yomimono) [inline reports/2017-05-28.md]: A link to http://canopy.mirage.io/Projects/test-tcpip (a higher-level view of what tests/testing-related work would be nice in tcpip) would fit well here.

**Comment by yomimono** [comment]
Just caught up after some time away via the 28 May report; thanks @avsm, it's very useful :)

File: reports/2017-05-28.md (hunk 2 above)
Comment (avsm) [inline reports/2017-05-28.md]: added in 32dcd200633030b5daec9fe0bfe4fc9a01eee3a9, thanks!

**Comment by djs55** [comment]
Thanks!

**Pull request merged**